### DISCUSSION TITLE
Add a `Map` DType

### DISCUSSION THREAD

**Discussion by doki23** [vortex-data/vortex#5844] [Feature Requests]
I notice that the `map` type does not appear in the `Logical Types`, do we have some plan of it?

**Comment by blaginin** [comment]
Our medium term plan is to support a Variant DType, allowing for arbitrary JSON-like nested structures with efficient shredding of common columns.

For now, it's possible to express Map as `List<Struct<String, V>>`

**Comment by lukekim** [comment]
We (Spice AI) need Map type support through the DataFusion integration.

There are a number of options:

## First-class DType

Does this just duplicate existing types/implementation?

## List/Struct Conversion

Example of basic List/Struct Conversion implementation: https://github.com/spiceai/vortex/pull/21/changes

List/Struct could be considered as stop gap for Variant support.

## An extension array

Pros/cons?

**Reply by joseph-isaacs** [reply]
This does seem like it should be an extension DType.

We are currently in the process of making these easy to extend. @gatesn
